src/loops/while_loops/hard/q6/MainTest.java

```diff
@@ -0,0 +1,54 @@
+package loops.while_loops.hard.q6;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    // Parsons
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "3";
+        return new Clause[]{
+                new StringLiteral("Enter a number:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() { Factorial.main(new String[0]); }
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of(0, 0, 3), Arguments.of(10, 3628800, 3), Arguments.of(5, 120, 3), Arguments.of(1, 1, 3), Arguments.of(12, 479001600, 3), Arguments.of(-6, -1, 1));
+    }
+
```

Comment:
I think you should have a separate test called something like `printsErrorMessageForInvalidInput()` so that in the end, the student would get the output of 
```
What went well:
- Calculates Factorial Correctly
- Prints Error Message ForI nvalidI nput
```